src/main/scala/ir/eval/InterpretTrace.scala

```diff
@@ -63,7 +64,7 @@ def interpretWithTrace[I](
   innerInterpreter: Effects[I, InterpreterError],
   innerInitialState: I
 ): (I, Trace) = {
-  InterpFuns.interpretProg(tracingInterpreter(innerInterpreter))(p, (innerInitialState, Trace(List())))
+  InterpFuns.interpretProg(tracingInterpreter(innerInterpreter))(p, (innerInitialState, Trace(Vector())))
```

Comment:
this and other places can use `Trace.empty`. or, a default value should be added to `Trace` to allow `Trace()`.